org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -239,10 +249,15 @@ private void appendGuessingCompletion(StringBuilder buffer, CompletionProposal p
 			}
 
 			char[] argument = parameterNames[i];
-
-			buffer.append("{{");
+			if(supportSnippets){
+				buffer.append("${");
+				buffer.append(Integer.toString(i+1));
+				buffer.append(":");
+			}
 			buffer.append(argument);
```

Comment:
Actually it makes sense that if a client does not support both snippetStrings and signature help we fall back to completing method names without parameters. I will also try to see if I have any easy way to filter overloaded methods.